lib/User/UserEntry.php

```diff
@@ -274,8 +274,9 @@ public function getHome() {
 		}
 
 		// TODO use OutOfBoundsException and https://github.com/owncloud/core/pull/28805
+		$enforce = $this->config->getAppValue('user_ldap', 'enforce_home_folder_naming_rule', true);
```

Comment:
I just want consistency with the rest of similar checks done in core:
```

  $tmpl = new Template('files_external', 'settings');
  $enabled = ($this->config->getAppValue('core', 'enable_external_storage', 'no') === 'yes'
…  
  $tmpl->assign('allowUserMountSharing', $this->config->getAppValue('core', 'allow_user_mount_sharing', 'yes') === 'yes');
  return $tmpl;
}
```
If you think using `filter_var` is better, open a ticket to use it across ownCloud and raise priority.

I'd prefer to avoid having to check for "yes" / "no" in one place, "true" / "false" in other, "on" / "off" in a third one and "1" / "0" in another. The behaviour will feel buggy if one configuration key supports "on" while the rest not.

Reply:
This should anyway be abstracted into a config class - `$class->isExternalStorageEnabled()` which returns a `bool` -> one single type hinted proper point. Classes should never care about details how they can check for `true/false`

Reply:
should be abstracted away in core ... I agree, using `filter_var` in this app makes it more robust against users setting config values via occ, [as documented](https://doc.owncloud.org/server/10.0/admin_manual/configuration/user/user_auth_ldap.html#special-attributes) eg for the home folder naming rule, which uses `--value=1` for whatever reason. Admins might set it to whatever, so I think making the check more robust is as good as the app can get right now.